hibernate-core/src/main/java/org/hibernate/cfg/JdbcSettings.java

```diff
@@ -471,6 +471,22 @@ public interface JdbcSettings extends C3p0Settings, ProxoolSettings {
 	 */
 	String CONNECTION_HANDLING = "hibernate.connection.handling_mode";
 
+	/**
+	 * Whether access to JDBC {@linkplain java.sql.DatabaseMetaData metadata} during bootstrap is allowed.
+	 * <p/>
+	 * Typically, Hibernate accesses this metadata to understand the capabilities of the underlying
+	 * database to help minimize needed configuration.  Disabling this access means that only explicit
+	 * settings are used.  At a minimum, the Dialect to use must be specified using either the {@value #DIALECT}
+	 * or {@value JdbcSettings#JAKARTA_HBM2DDL_DB_NAME} setting.
```

Comment:
I think we should probably start trying to guess the dialect from the JDBC URL, at least for the major platforms we support. That's what we do in Reactive, IIRC.

Reply:
Another thing: I would tell them to not only specify the dialect, but also the database version.

Reply:
> I think we should probably start trying to guess the dialect from the JDBC URL

Wait, I think we actually _do_ do this but I guess only when using `DriverManager` directly.

Reply:
> Another thing: I would tell them to not only specify the dialect, but also the database version.

That's reasonable to notate in the Javadoc.

> Wait, I think we actually do do this but I guess only when using DriverManager directly.

Right, when the URL would be specified anyway.  I think ex[ecting users who are using data sources to specify URL just for the purpose of resolving the Dialect to use is silly.  If they are (1) using a datasource and (2) want to disable metadata access on bootstrap - specifying the Dialect to use via either `DIALECT` or `JAKARTA_HBM2DDL_DB_NAME` is more than reasonable.

Reply:
> That's reasonable to notate in the Javadoc.

Yup, that's what I mean.

>  I think expecting users who are using data sources to specify URL just for the purpose of resolving the Dialect to use is silly. 

Yeah sure, agreed.